Style for `ccorres` statements

michaelmcinerney opened the pull request.

corlewis reviewed (commented):
The actual content looks good to me. However, I can't be sure without opening it up myself, but do you need all of the fake constants? I would have thought you would just need `ccorres`, `UNIV`, `c_guard`, `ptr_select`, and maybe `ptr` and `cond`. Couldn't the `ccorres` arguments be bound variables?

Also, you shouldn't need the axiomatizations unless you explicitly refer to the definition lemma.

michaelmcinerney commented:
> The actual content looks good to me. However, I can't be sure without opening it up myself, but do you need all of the fake constants. I would have thought you would just need `ccorres`, `UNIV`, `c_guard`, `ptr_select`, and maybe `ptr` and `cond`. Couldn't the `ccorres` arguments be bound variables?
> 
> Also, you shouldn't need the axiomatizations unless you explicitly refer to the definition lemma.

Very good point. None of the constants is needed. I've removed them, and adjusted things in the fake lemmas to make the schematic variable names nicer

Xaphiosis commented on docs/Style.thy:
```diff
@@ -323,6 +323,59 @@ lemma my_hoare_triple_lemma:
+  oops
+
+lemma longer_ccorres_example:
+  "ccorres rrel xf
```
something to consider: what about complex return relations? I think they should nearly all fit, I haven't found anything more complex than:

`ccorres  (cfault_rel2 ¤ dc) (liftxf errstate id (K ()) ret__unsigned_long_')`

michaelmcinerney replied:
Sure, I'll put a `longer_rrefl` and `longer_xf` on separate lines in this example

Xaphiosis commented on docs/Style.thy:
```diff
@@ -323,6 +323,59 @@ lemma my_hoare_triple_lemma:
+  The concrete guard will often be simply @{term UNIV}, or an intersection of terms of the form
+  @{term "\<lbrace>\<acute>pointer = cond\<rbrace>"}, which supersedes the set-builder notation. Note that the
+  semantically redundant form @{term "UNIV \<inter> \<lbrace>\<acute>pointer = cond\<rbrace>"} should no longer be necessary,
+  but may still be required in some places.\<close>
```
I'd tweak this a little bit. We want to indicate that we don't want people using it unless they are forced to by legacy tactics/methods or related issues.

Xaphiosis commented on docs/Style.thy:
```diff
@@ -323,6 +323,59 @@ lemma my_hoare_triple_lemma:
+  Here are some examples.\<close>
+
+lemma short_ccorres_example:
+  "ccorres rrel xf short_abs_guard short_conc_guard hs short_abs_fn short_conc_fn"
```
This is off-topic slightly, but `conc` always makes me think of "concurrent". `concr` might be less ambiguous, but up to you.

lsf37 reviewed (approved):
Looks good from my side!

Xaphiosis commented:
@michaelmcinerney : any reason this wasn't merged?

michaelmcinerney commented:
> @michaelmcinerney : any reason this wasn't merged?

No good reason. I've been meaning to ask whether it's fine to merge. I'll try today
